src/openvic-simulation/map/MapDefinition.hpp

```diff
@@ -20,6 +21,16 @@ namespace OpenVic {
 	struct BuildingTypeManager;
 	struct ModifierManager;
 
+	struct RiverSegment {
+		friend struct MapDefinition;
+
+	private:
+			const uint8_t PROPERTY(size);
+			const std::vector<ivec2_t> PROPERTY(points);
+
+			RiverSegment(uint8_t new_size, std::vector<ivec2_t> new_points);
```

Comment:
```suggestion
		const uint8_t PROPERTY(size);
		std::vector<ivec2_t> PROPERTY(points);

		RiverSegment(uint8_t new_size, std::vector<ivec2_t>&& new_points);

	public:
		RiverSegment(RiverSegment&&) = default;
```
- Make `points` non-const so it can be moved (const vectors can only be copied).
- For `new_points` to be a r-value reference/moved.
- Add a public default move constructor
- Indent back a tab to be one past private.